### ISSUE TITLE
Does it compress the uploaded image in default?

### ISSUE THREAD

**Issue by badbye** [gradio-app/gradio#14]
I find the size of the received base64 encoded image, is much smaller than the original image.
Furthermore, I find the following code in the `image_upload.js` file.
```
resizeImage.call(this, data, 600, 600, function(image_data) {
      io.image_data = image_data
      io.target.find(".image_preview").attr('src', image_data);
      if (update_editor) {
        io.tui_editor.loadImageFromURL(io.image_data, 'input').then(function (sizeValue) {
          io.tui_editor.clearUndoStack();
          io.tui_editor.ui.activeMenuEvent();
          io.tui_editor.ui.resizeEditor({ imageSize: sizeValue });
        });
      }
```
So the maximum size of the received image is 600 * 600, right? Is it possible to undo this compression?

**Comment by badbye** [comment]
I believe I am right. It did do compression, the limitation is 300*300 due to the `submit` function: 
```javascript
submit: function() {
    if (this.state == "IMAGE_LOADED") {
      resizeImage.call(this, this.image_data, 300, 300, function(image_data) {
        this.io_master.input(this.id, image_data);
      })
    }
  },
```

So firstly, the size limitation of the preview step is 600*600.
Secondly, the size limitation of the submit step is 300*300.
If I want to upload the original file exactly, I need to change both of them.

Sometimes the size of the file does matter, it is better to offer an API to let the uploaded file keep its width and height. Hope you agree with me.

**Comment by dawoodkhan82** [comment]
> I believe I am right. It did do compression, the limitation is 300*300 due to the `submit` function:
> 
> ```js
> submit: function() {
>     if (this.state == "IMAGE_LOADED") {
>       resizeImage.call(this, this.image_data, 300, 300, function(image_data) {
>         this.io_master.input(this.id, image_data);
>       })
>     }
>   },
> ```
> 
> So firstly, the size limitation of the preview step is `600*600`.
> Secondly, the size limitation of the submit step is `300*300`.
> If I want to upload the original file exactly, I need to change both of them.
> 
> Sometimes the size of the file does matter, it is better to offer an API to keep the original width and height. Hope you agree with me.

Hey badbye, I wanted to reach out to see if you would be willing to hop on a quick call with us. We wanted to talk to some users of Gradio-UI ad get your feedback. What e-mail is best to reach you?

**Comment by badbye** [comment]
> Hey badbye, I wanted to reach out to see if you would be willing to hop on a quick call with us. We wanted to talk to some users of Gradio-UI ad get your feedback. What e-mail is best to reach you?

Hi, I am glad to share my thoughts. Here is my email: eWFsZWlkdUAxNjMuY29t (use base64 to decode it).

**Comment by ranftlr** [comment]
I'm facing a similar issue. For dense prediction tasks (semantic segmentation, depth estimation, etc.) it would be helpful to get the image in its original shape from the API.

**Comment by abidlabs** [comment]
Makes sense, let's make this as an option in the input interface @aliabid94

**Comment by BhavanJ** [comment]
First of all very nice work with Gradio, it's so awesome. Immensely useful to create quick demos! Any updates regarding the issue of receiving image at original resolution?

I am facing a similar issue, in particular if the image contains text (and is of a larger dimensions > 300 x300), the received image has much lower quality (hard to read) and OCR fails badly on it. See the example below, uploaded image on left and received image on right. The text is very hard to read. Thanks!

<img width="834" alt="gradio_git_isssue" src="https://user-images.githubusercontent.com/22835023/89455093-9c509300-d716-11ea-847a-c8ae5c48790a.png">

**Comment by aliabid94** [comment]
This has now been fixed. There should be no automatic resizing anymore.

**Comment by aliabid94** [comment]
The latest pip package has the updated version.

**Comment by BhavanJ** [comment]
@aliabid94 thanks for getting back. I checked with version 1.0.6, if I don't mention the size of the input image, it still automatically resizes the image (I get in my python function) so that the max dimension of image is 600.

Another thing is if I mention a fixed larger image size like 1000 x 1200 in the input image definition, I do get image of the specified fixed dimension. But the quality of the image is degraded as shown in the figure in my earlier post, making it hard to read the text. This effects the predictions of the machine learning model I use. 

I tried multiple times. Can you please double check if the image compression and resizing is fixed?

**Comment by aliabid94** [comment]
@BhavanJ can you try doing a hard refresh from your browser to clear the cached versions of javascript files? In the future we'll version our static files so this won't be necessary.
Appreciate your use and kind words on Gradio 👍 :)

**Comment by BhavanJ** [comment]
@aliabid94 thanks again for getting back. I tried doing that, I also tried running on a different laptop, different browser but still facing same issue. Do you mind trying the below demo? I am just taking as input an image and returning it back along with the dimensions of the input image. And I added some example images. If you view the example images they are bigger than 600, while if you view the input and output image they are limited to 600 and also low resolution.

Demo linke - https://45585.gradio.app

**Comment by aliabid94** [comment]
You're right, fixing now. Thanks for the detailed bug report.

**Comment by aliabid94** [comment]
Okay fixed, please upgrade to gradio 1.0.7. Let me know if you have any more issues!

**Comment by BhavanJ** [comment]
Cool thanks a lot, let me try it out.

**Comment by BhavanJ** [comment]
I tried, works great so far. Thanks a lot for fixing this @aliabid94! This would be very helpful for any demos which need high resolution images.

**Comment by aliabid94** [comment]
Glad to hear it! Let us know anything else that could be improved :)

**Comment by mayayank95** [comment]
> Okay fixed, please upgrade to gradio 1.0.7. Let me know if you have any more issues!

I try to upgrade the radio but it still compresses the data...
